src/main/java/com/softserve/academy/spaced/repetition/service/impl/AccountServiceImpl.java

```diff
@@ -54,10 +55,21 @@ public LearningRegime getLearningRegime() throws NotAuthorisedUserException {
 
     @Override
     @Transactional
-    public void updateLearningRegime(LearningRegime learningRegime) throws NotAuthorisedUserException {
-        Account account = accountRepository.findOne(userService.getAuthorizedUser().getAccount().getId());
-        account.setLearningRegime(learningRegime);
-        accountRepository.save(account);
+    public void updateLearningRegime(String learningRegime) throws NotAuthorisedUserException, IllegalArgumentException {
+        boolean learningRegimeFound = false;
+        for (LearningRegime regime : LearningRegime.values()) {
```

Comment:
Good. However, if you want, you can use stream to have less code. Up to you. I will show